embassy-stm32/src/spi/mod.rs

```diff
@@ -623,11 +627,114 @@ impl<'d> Spi<'d, Async> {
 
     /// SPI read, using DMA.
     pub async fn read<W: Word>(&mut self, data: &mut [W]) -> Result<(), Error> {
+        #[cfg(not(spi_v3))]
```

Comment:
it's simpler to cfg the entire `read` function instead of having it call two functions that need cfg'ing anyway